### ISSUE TITLE
Combine Applications with the same configuration on Auth Server Web UI

### ISSUE THREAD

**Issue by pschisa** [gravitational/teleport#5580]
### What
Currently, if multiple Teleport application service nodes are created to reference the same Teleport application for redundancy/HA, they are listed in the same auth server as separate applications. 

Request is to have multiple instances of the same application only show as a single icon in the Web UI to allow for redundant application service nodes to not clutter the application UI.

### How
If all parameters of an application match, they could be treated as the same application for display purposes in the Web UI / CLI and select either service based on which is available at the time when launched from the Web UI or CLI.

### Why
A cleaner way to create redundancy for the application service.

### Which platform(s)
- [ x] Teleport Cloud
- [x] Teleport Enterprise

### Which component(s)
- [ X] Application Access
- [X ] Web Clients

**Comment by programmerq** [comment]
Another concern that came up is that it should be possible to have redundancy so any one app_service instance won't be a single point of failure for that application. I'd say that a related request would be that kube_service and db_service should also support multiple instances for redundancy sake.

**Comment by tpecora-cpi** [comment]
Yes it would be great to apply this to all teleport services [app, db, ssh, k8].  If verifying all parameters on the agent configuration match, is too complicated or may cause false positives.  For example, say the endpoint for an app is http://localhost:8080 and this exists on two agents.... are they the same (and can be combined into one UI/CLI entry?  Maybe, maybe not?  Another option might be to add another yaml parameter where we specifically state **priority: 1** (where the lower the number sets the higher priority).  Example:
-------------------------------------------
TELEPORT-AGENT-1
  - name: "web-app"
    # URI and port of application.
    uri: "https://web-app.company.local"
    priority: 1
    labels:
      company: C1
      environment: PROD
      infra: DATA
      region: us-west-2
      system: web-app
-------------------------------------------
TELEPORT-AGENT-2
  - name: "web-app"
    # URI and port of application.
    uri: "https://web-app.company.local"
    priority: 2
    labels:
      company: C1
      environment: PROD
      infra: DATA
      region: us-west-2
      system: web-app
